src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/Extensions/DSC/GetAzureVMDscExtensionStatus.cs

```diff
@@ -0,0 +1,242 @@
+
+            if (CurrentDeploymentNewSM == null)
+            {
+                WriteWarning(
+                    string.Format(CultureInfo.CurrentUICulture, Resources.NoDeploymentFoundInService, Service));
+                return;
+            }
+            
+            var vmDscStatusContexts = GetVirtualMachineDscStatusContextList(CurrentDeploymentNewSM);
+            if (vmDscStatusContexts == null || vmDscStatusContexts.Count < 1)
+            {
+                WriteWarning(Resources.ResourceExtensionReferenceCannotBeFound);
+            }
+            WriteObject(vmDscStatusContexts, true);
+        }
+
+        /// <summary>
+        /// Retrieves service name from the cmdlet's service name or virtual machine parameter 
+        /// </summary>
+        /// <param name="serviceName">Name of the cloud service for DSC Extension Status</param>
+        /// <param name="vm">Name of the VM in a cloud service for DSC Extension Status </param>
+        internal void GetService(String serviceName, IPersistentVM vm)
+        {
+            if (!string.IsNullOrEmpty(serviceName))
+            {
+                Service = serviceName;
+            }
+            else
+            {
+                //get the service name from the VM object
+                var vmRoleContext = vm as PersistentVMRoleContext;
+                if (vmRoleContext == null)
+                    return;
+
+                Service = vmRoleContext.ServiceName;
+                VmName = vmRoleContext.Name;
+            }
+        }
+
+        /// <summary>
+        /// Retrieves deployment information for a cloud service from service api's 
+        /// </summary>
+        internal void GetCurrentDeployment()
+        {
+            InvokeInOperationContext(() =>
+            {
+                try
+                {
+                    if (string.IsNullOrEmpty(Service))
+                        return;
+                    
+                    CurrentDeploymentNewSM = ComputeClient.Deployments.GetBySlot(Service, NSM.DeploymentSlot.Production);
+                    GetDeploymentOperationNewSM = GetOperationNewSM(CurrentDeploymentNewSM.RequestId);
+                    WriteVerboseWithTimestamp(Resources.GetDeploymentCompletedOperation);
+                }
+                catch (CloudException ex)
+                {
+                    if (ex.Response.StatusCode != HttpStatusCode.NotFound)
+                    {
+                        throw;
+                    }
+                }
+            });
+        }
+
+        /// <summary>
+        /// Retrieves dsc extension status for all virtual machine's in a cloud service or a given virtual machine from the deployment object
+        /// </summary>
+        /// <param name="deployment">Deployment that exists in the service</param>
+        /// <returns></returns>
+        internal List<VirtualMachineDscExtensionStatusContext> GetVirtualMachineDscStatusContextList(NSM.DeploymentGetResponse deployment)
```

Comment:
Are you referring to creating a separate common class for the business logic for all DSC cmdlets or just this cmdlet? The logic and data manipulation done in this cmdlet is not going to be used by elsewhere. So, I feel that putting it in a different class is not going to be of any use.